pkg/storage/iterator_test.go

```diff
@@ -306,4 +554,18 @@ func Test_newBatchChunkIterator(t *testing.T) {
 		})
 	}
 
+	for name, tt := range tests {
+		tt := tt
+		t.Run(fmt.Sprintf("large-batchsize/%s", name), func(t *testing.T) {
+			it := newBatchChunkIterator(context.Background(), tt.chunks, 1000, newMatchers(tt.matchers), nil, newQuery("", tt.start, tt.end, tt.direction))
```

Comment:
Could you explain the benefit of the large batchsize tests? Additionally, would it be better to use `len(tt.chunks)` instead of `1000` here for extensibility to ensure the the batch is always large enough to hold all the chunks?

Reply:
To be honest I don't know whether these tests are needed or not. I added these tests for debugging purposes since the bug described in #1187 don't happen when the batchsize is big enough to hold all of the chunks. I can remove those if deemed unnecessary.